File: Java/Easy/WrongAnswer/greatest-english-letter-in-upper-and-lower-case_8.java
class Solution {
    public String greatestLetter(String s) {
        
        char[] arr = s.toCharArray();
        Arrays.sort(arr);
        String res = "";
        for (int i = 0; i < s.length() - 1; i++) {
            if (arr[i] == arr[i + 1]) continue;
            if (arr[i] + 32 == arr[i + 1]) {
                res = Character.toString(arr[i + 1]).toUpperCase();
            }
        }
        return res;
    }
}

File: Rust/Easy/CompileError/greatest-english-letter-in-upper-and-lower-case_8.rs
impl Solution {
    pub fn greatest_letter(s: String) -> String {
        
        let mut s = s.as_bytes();
        s.sort();
        let mut it = s.iter();
        let mut prev = it.next();
        for c in it {
            if let Some(prev) = prev {
                if (prev.is_ascii_lowercase() && c.is_ascii_uppercase() && prev.to_ascii_uppercase() == *c) || (prev.is_ascii_uppercase() && c.is_ascii_lowercase() && prev.to_ascii_lowercase() == *c) {
                    return String::from_utf8(vec![*c]).unwrap();
                }
            }
            prev = Some(c);
        }
        "".to_string()
    }
}